docs/02.API-REFERENCE.md

```diff
@@ -101,6 +101,39 @@ created by API functions has the error flag set.
 typedef uint32_t jerry_value_t;
 ```
 
+## jerry_context_data_header_t
+
+**Summary**
+
+Structure that holds a single user context item, identified by its manager. It is intended to serve as the base for
+larger structures that need to be allocated at runtime.
+
+**Prototype**
+
+```c
+typedef struct jerry_context_data_header
+{
+  struct jerry_context_data_header *next_p; /**< pointer to next context item */
+  const struct jerry_context_data_manager *manager_p; /**< manager responsible for deleting this item */
+} jerry_context_data_header_t;
```

Comment:
Though that would mean calling `init_cb` twice:

```c
typedef struct
{
  int something;
  int something_else;
} my_data_t;

size_t
my_data_init(void *context_data_pp)
{
  if (context_data_p)
  {
    (( my_data_t *) context_data_p)->something = 5;
    (( my_data_t *) context_data_p)->something_else = 12;
  }
  return sizeof (my_data_t);
}
```
and in the `jerry_get_context_data ()`
```c
/* First call to init_cb () determines how much it needs */
size_t size_needed = manager_p->init_cb (NULL);
jerry_context_data_header_t *data_p =
jmem_heap_alloc_block (sizeof (jerry_context_data_header_t) + size_needed);

data_p->next_p = JERRY_CONTEXT (context_data_p);
data_p->manager_p = manager_p;
JERRY_CONTEXT (context_data_p) = data_p;

void *user_data_p = &((char *) data_p)[sizeof (jerry_context_data_header_t)];

/* Second call to init_cb () has a non-NULL pointer as the parameter,
 * so that means initialize the memory at the pointer */
manager_p->init_cb (user_data_p);
...
return user_data_p;
```